test/Microsoft.AspNetCore.Mvc.Formatters.Json.Test/JsonInputFormatterTest.cs

```diff
@@ -27,6 +27,7 @@ public class JsonInputFormatterTest
         private static readonly ObjectPoolProvider _objectPoolProvider = new DefaultObjectPoolProvider();
         private static readonly JsonSerializerSettings _serializerSettings = new JsonSerializerSettings();
 
+#pragma warning disable CS0618
```

Comment:
Narrow this further throughout the PR; place the `#pragma`s to surround just the calls to `[Obsolete]` constructors.